### PR TITLE
feat: error messages for Presto connections 

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Add specific error messages for Presto.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before we had only generic messages:

Screen Shot 2021-04-15 at 7 35 42 AM

Screen Shot 2021-04-15 at 7 35 48 AM

Screen Shot 2021-04-15 at 7 36 03 AM

Now:

Screen Shot 2021-04-15 at 10 04 27 AM

Screen Shot 2021-04-15 at 10 09 15 AM

### TEST PLAN

Added unit tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docs/src/pages/docs/Miscellaneous/issue_codes.mdx [hunk 1]
```diff
@@ -166,3 +166,12 @@ Either the database is spelled incorrectly or does not exist.
...
 Either the database was written incorrectly or it does not exist. Check that it was typed correctly.
+
+
+## Issue 1016
+
+```
+The schema was deleted or renamed in the database.
+```
+
+Either the schema was removed or renamed. Check that the schema is typed correctly and exists.
```
Comment (AAfghahi) [inline docs/src/pages/docs/Miscellaneous/issue_codes.mdx:177]:
```suggestion
The schema was either removed or renamed. Check that the schema is typed correctly and exists.
```
just a nit

File: superset/databases/commands/exceptions.py [hunk 2]
```diff
@@ -130,7 +130,8 @@ class DatabaseTestConnectionDriverError(CommandInvalidError):
     message = _("Could not load database driver")
 
 
-class DatabaseTestConnectionUnexpectedError(CommandInvalidError):
+class DatabaseTestConnectionUnexpectedError(SupersetErrorsException):
+    status = 422
```
Comment (AAfghahi) [inline superset/databases/commands/exceptions.py:134]: Where could I learn what status codes signify what?

File: tests/db_engine_specs/presto_tests.py [hunk 3]
```diff
@@ -829,6 +830,193 @@ def test_extract_error_message_general_exception(self):
...
+            )
+        ]
+
+        msg = b"Access Denied: Invalid credentials"
+        result = PrestoEngineSpec.extract_errors(Exception(msg), {"username": "alice"})
+        assert result == [
+            SupersetError(
+                message='Either the username "alice" or the password is incorrect.',
+                error_type=SupersetErrorType.CONNECTION_ACCESS_DENIED_ERROR,
+                level=ErrorLevel.ERROR,
```
Comment (AAfghahi) [inline tests/db_engine_specs/presto_tests.py:928]: I think this one needs to be reformatted to reflect the newer language? Unless we merge this one in first, then merge the mssql changes.

File: superset/databases/commands/exceptions.py (hunk 2 above)
Comment (lyndsiWilliams) [reply superset/databases/commands/exceptions.py:134]: I use the Mozilla docs for reference, unless we've got custom status codes:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Comment (hughhhh) [reply superset/databases/commands/exceptions.py:134]: https://httpstatuses.com is good too

**Comment by AAfghahi** [comment]
Did you add this to:
superset-frontend/src/setup/setupErrorMessages.ts ?

**Comment by betodealmeida** [comment]
> Did you add this to:
> superset-frontend/src/setup/setupErrorMessages.ts ?

Good point, I forgot.

File: superset/databases/commands/test_connection.py [hunk 4]
```diff
@@ -108,7 +111,8 @@ def run(self) -> None:
                 action=f"test_connection_error.{ex.__class__.__name__}",
                 engine=database.db_engine_spec.__name__,
             )
-            raise DatabaseTestConnectionUnexpectedError(str(ex))
+            errors = database.db_engine_spec.extract_errors(ex, context)
```
Comment (hughhhh) [inline superset/databases/commands/test_connection.py:114]: nice i was just about to add this to my google api PR

**Review by hughhhh — APPROVED** [review]

**Pull request merged**